### DISCUSSION TITLE
How to consolidate all windows spreading in multiple virtual desktops into a specific virtual desktop and restore them?

### DISCUSSION THREAD

**Discussion by zanhsieh** [rxhanson/RectanglePro-Community#580] [Q&A] [answered]
Hello,

As title. Just curious if it's possible with Rectangle Pro. My scenario is, I owned a VITURE Pro XR, an AR glasses:

https://pro.viture.com/

In MacOS, the screen it projects as an external monitor no matter what kind of multiple virtual monitor layout, e.g. 3 virtual monitor side-by-side, a super-wide monitor, a super-wide curve monitor. So when hit the road, I need to move all windows originally spreading across multiple desktops into the VITURE Pro XR desktop. Once I go back to the stationary, all windows have to get distributed back to their own virtual desktop.

**Comment by rxhanson** [comment]
Thanks for the feedback. Unfortunately, Apple has not released public APIs for moving windows across Spaces (AKA virtual desktops). I can revisit this idea if this changes.

**Reply by zanhsieh** [reply]
Hello @rxhanson ,

Thanks for swift reply. How about consolidate all windows into the current focused virtual desktop? Is it possible?

**Reply by rxhanson** [reply, marked as answer]
Unfortunately not. Hopefully at some point Apple releases an API that would allow this.
